galsim/lsst/lsst_wcs.py

```diff
@@ -0,0 +1,1153 @@
+
+        rotation_angle = 180 degrees means north is -x and east is -y
+        """
+        return self._rotation_angle
+
+
+    def pupilCoordsFromPoint(self, point):
+        """
+        Convert from RA, Dec into coordinates on the pupil
+
+        inputs
+        ------------
+        point is a CelestialCoord (or a list of CelestialCoords) indicating
+        the positions to be transformed
+
+        outputs
+        ------------
+        The x and y coordinates on the pupil in radians
+        """
+
+        if not hasattr(point, '__len__'):
+            pt = self._pointing.project(point, projection='gnomonic')
+            x = pt.x
+            y = pt.y
+        else:
+            x = []
+            y = []
+            for pp in point:
+                pt = self._pointing.project(pp, projection='gnomonic')
+                x.append(pt.x)
+                y.append(pt.y)
+            x = np.array(x)
+            y = np.array(y)
+
+        return (x*self._cos_rot - y*self._sin_rot)*galsim.arcsec, \
+               (x*self._sin_rot + y*self._cos_rot)*galsim.arcsec
+
+
+    def pupilCoordsFromFloat(self, ra, dec):
+        """
+        Convert from RA, Dec into coordinates on the pupil
+
+        Note: this method just reimplements the PALPY method Ds2tp
+        with some minor adjustments for sign conventions.
+
+        inputs
+        ------------
+        ra is in radians.  Can be a list.
+
+        dec is in radians.  Can be a list.
+
+        outputs
+        ------------
+        The x and y coordinates on the pupil in radians
+        """
+
+        dra = ra - self._pointing.ra/galsim.radians
+        cradif = np.cos(dra)
+        sradif = np.sin(dra)
+        sdec = np.sin(dec)
+        cdec = np.cos(dec)
+        denom = sdec * self._sin_dec + cdec * self._cos_dec * cradif
+        xx = cdec * sradif/denom
+        yy = (sdec * self._cos_dec - cdec * self._sin_dec * cradif)/denom
+        xx *= -1.0
+        return xx*self._cos_rot - yy*self._sin_rot, xx*self._sin_rot + yy*self._cos_rot
+
+
+    def raDecFromPupilCoords(self, x, y):
+        """
+        Convert pupil coordinates in radians to RA, Dec in radians
+
+        Note: this method just reimplements the PALPY method Dtp2s
+        with some minor adjustments for sign conventsion
+
+        inputs
+        ------------
+        x is the x pupil coordinate in radians
+
+        y is the y pupil coordinate in radians
+
+        ouputs
+        ------------
+        RA and Dec in radians as lists of floats (or individual floats if only one
+        set of x, y was passed in)
+        """
+
+        x_g = x*self._cos_rot + y*self._sin_rot
+        y_g = -1.0*x*self._sin_rot + y*self._cos_rot
+
+        x_g *= -1.0
+
+        denom = self._cos_dec - y_g * self._sin_dec
+        d = np.arctan2(x_g, denom) + self._pointing.ra/galsim.radians
+        ra = d%(2.0*np.pi)
+        dec = np.arctan2(self._sin_dec + y_g * self._cos_dec, np.sqrt(x_g*x_g + denom*denom))
```

Comment:
Likewise, can this use the `deproject` function?  Maybe cleaner than reimplementing this functionality here.